apiserver/client/status.go

```diff
@@ -589,6 +589,30 @@ type stateAgent interface {
 	Status() (state.Status, string, map[string]interface{}, error)
 }
 
+// processUnitAgent retrieves status information for both unit and unitAgents.
+func processUnitAgent(unit *state.Unit) (agent api.AgentStatus,
```

Comment:
UnitAgent can be ambiguous - can we rename this to something like processUnitAndAgentStatus()